core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -286,6 +289,93 @@ public void createCatalog(CatalogName catalogName, ConnectorName connectorName,
         }
     }
 
+    @Override
+    public void renameCatalog(CatalogName catalogName, CatalogName newCatalogName)
+    {
+        requireNonNull(catalogName, "catalogName is null");
+
+        catalogsUpdateLock.lock();
+        try {
+            checkState(state != State.STOPPED, "ConnectorManager is stopped");
+
+            if (!activeCatalogs.containsKey(catalogName)) {
+                throw new TrinoException(NOT_FOUND, format("Catalog '%s' does not exist", catalogName));
+            }
+            if (activeCatalogs.containsKey(newCatalogName)) {
+                throw new TrinoException(ALREADY_EXISTS, format("Catalog '%s' already exists", newCatalogName));
+            }
+
+            createCatalogLikeInternal(catalogName, newCatalogName, ImmutableMap.of());
+            catalogStore.removeCatalog(catalogName);
+            activeCatalogs.remove(catalogName);
```

Comment:
how do we ensure this is "transactional"?

Reply:
the replace would probably also allow implementing `ALTER CATALOG ... RENAME` properly instead of as `DROP` + `CREATE`.

Let's skip for now and discuss it later.